app/main/views/users.py

```diff
@@ -271,24 +282,44 @@ def export_users_for_framework(framework_slug):
 
     suppliers_with_a_complete_service = frozenset(framework.get_supplier_ids_for_completed_service())
 
-    supplier_id_published_service_count = dict(db.session.query(
-        Service.supplier_id,
-        func.count(Service.id)
-    ).filter(
-        Service.status == 'published',
-        Service.framework_id == framework.id
-    ).group_by(
-        Service.supplier_id
-    ).all())
+    lots_ids = db.session.query(FrameworkLot.lot_id).filter(FrameworkLot.framework_id == framework.id).all()
+    lots = db.session.query(Lot).filter(Lot.id.in_(lots_ids)).all()
+    published_service_count_by_supplier_and_lot = {}
+
```

Comment:
These 2 queries can be minimised to a single query with a join like so:
```
db.session.query(FrameworkLot.lot_id, Lot.slug).join(Lot, FrameworkLot.lot_id == Lot.id).filter(
        FrameworkLot.framework_id == 4
    ).distinct().all()
```

Joins are generally considered more performant than creating multiple queries (there are some exceptions).
https://dba.stackexchange.com/a/43018

Reply:
Below you'd then have the for statement like so:

```
for lot_id, slug in lots:
    ...
```